### ISSUE TITLE
[Q&A] Technikai érvénytelenítés vs HASH hitelesítésű számla

### ISSUE THREAD

**Issue by topybear** [nav-gov-hu/Online-Invoice#1019]
Ezen esetekben a hitelességet a NAV biztosítja. Felmerült most, hogy mi van akkor, ha egy ilyen számlát letöltünk, befogadunk, majd a kibocsátó csinál rájuk egy technikai érvénytelenítést és erről mondjuk nem küld értesítőt vagy az valamiért spam-ben landol. Ha ugyanolyan számlaszámmal csinál új számlát, akkor rendben van, mert azt látni fogjuk a következő letöltésnél. De mi van akkor, ha ilyen nincs? Akkor lesz egy olyan számlánk amiről mi azt gondoljuk, hogy létezik és hiteles, valójában meg nem is létezik.

Van esetleg a NAV-nak adata arról, hogy mennyire gyakori az az eset, amikor technikai érvénytelenített számla nem kerül ismételten, új adattartalommal feltöltésre?

Az iktatás részen beindult ezen a vergődés, de ha ez nem jellemző szituáció, akkor ezzel le tudnám ütni. Egyébként meg van a "kérdezzük le állandóan az összes ilyen számlát", de azt nagyon nem szeretném ( a szervert is fölöslegesen terhelnénk). Az elegáns az lenne, ha a letöltések közt az érvénytelenített számlaszámokat is megkapnánk, mert akkor tudnánk kezelni, hogy ha befogadott HASH-es van köztük, akkor félre kell verni a harangokat :), de ilyen egyelőre nincs.

Esetleg valaki foglalkozott már ezzel a problémával?

**Comment by omachtandras** [comment]
Minket is érdekelne egy olyan megoldás, ahol ha egy partner technikai érvényetelenítést végez egy olyan számlára, amin mi vagyunk a másik oldal, akkor arról információt kaphassunk, mert lehet, hogy ezzel el kellene indítani egy feladatot a cégen belül....

**Comment by programozas** [comment]
Minket is érdekel...

**Comment by szkomaromi-contactplus** [comment]
Ha ez az az eset, amikor a completenessIndicator = ture, akkor a leírás 2.4.1.10 pontja alapján nem lehet technikailag érvényteleníteni ilyen adatszolgáltatásokat. 

> 10. A completenessIndicator = true jelzésű adatszolgáltatásokat, ahol az adatszolgáltatás maga az
> elektronikus számla, nem lehetséges technikai érvényteleníteni. Ezt külön üzleti validáció figyeli,
> részletek a „Hibakezelés” fejezetben.

De ha lenne valami egyszerűbb módja a technikai érvénytelenített számlák bejövő oldali lekérdezésének, az nagyon hasznos lenne.

**Comment by topybear** [comment]
> 

Nem, ezek a HASH hitelesítéső számlák, azoknál a completenessIndicator = false.

**Comment by topybear** [comment]
Kicsit elült ez a téma :)
Jelenleg ami megoldási lehetőségeket látok:
1. NAV a HASH hitelesítésű számlák esetén is tiltani fogja a technikai érvénytelenítést - ez olcsó, egyszerű és megoldja a problémát
2. A számla letöltésekben valahogy megjelenításre kerülnek az adott időszakban érvénytelenített HASH-es számlák; fejlesztés kell mindkét oldalon, bonyolult
3. Minden így marad, a napi letöltéseknél az összes befogadott HASH-es számlát lekérdezzük, hogy élnek-e még: fejlesztés a mi oldalunkon, feleslegesen terheljük a szervert, hosszabb távon, sok ilyen számlánál a fejünkre nő az egész

Én mindenképp az első megoldást tartanám jónak, azaz ha ezeket ugyanúgy nem lehetne technikai érvényteleníteni, mint amikor az adatszolgáltatás maga a számla.
